pinctrl/gpio.go

```diff
@@ -38,8 +40,30 @@ type gpioPin struct {
 	swPwmCancel func()
 }
 
+// GPIOPin is the struct defining a GPIOPin.
+type GPIOPin struct {
+	*gpioPin
```

Comment:
why do you need to wrap this? can we just use the type directly?

Reply:
I needed have an exported the struct in order to use it in the pi5 package. It was quicker to wrap it but I can just make the underlying struct exported if people prefer

Reply:
I would prefer to just make the struct public but don't feel strongly

Reply:
oh i just switched it!